server/server/route/auth.go

```diff
@@ -189,15 +188,15 @@ func randNonce(c echo.Context) (string, error) {
 	}
 
 	returnURL := c.QueryParam("returnUrl")
-	u, err := url.Parse(returnURL)
-	if err != nil {
-		return "", fmt.Errorf("invalid returnUrl: %w", err)
-	}
+	// u, err := url.Parse(returnURL)
+	// if err != nil {
+	// 	return "", fmt.Errorf("invalid returnUrl: %w", err)
+	// }
 
 	// Reject redirects to other hosts, our use case only needs to handle local redirects.
-	if u.Host != "" && u.Host != c.Request().Host {
-		return "", fmt.Errorf("invalid returnUrl: does not match expected host %s", c.Request().Host)
-	}
+	// if u.Host != "" && u.Host != c.Request().Host {
+	// 	return "", fmt.Errorf("invalid returnUrl: does not match expected host %s", c.Request().Host)
+	// }
```

Comment:
This is probably a controversial change. But we need to be able to redirect from port 3000 (UI) -> 8081 (ui-server) -> 8889 (oidc-server) -> 3000 (UI).